### PR TITLE
fix(dogfood): fix the issue #164 regarding the voting power not update

### PR DESCRIPTION
**Pull request by TimmyExogenous**
## Description

close #164 

This PR fixes issue #164 by updating the Dogfood AVS information when the parameter in the Dogfood module is updated.

## Summary by CodeRabbit

- **New Features**
	- Enhanced update process for AVS information with authorization checks and error handling.
	- Introduced new error variables for improved error reporting related to AVS by chain ID.
	- Added a new method to the AVSKeeper interface for managing AVS updates.
	- Improved transaction size validation logic in the transaction handling process.

- **Bug Fixes**
	- Improved parameter handling in command-line interface functions for better type consistency.

- **Style**
	- Streamlined code formatting and organization in various test files for improved readability.

- **Chores**
	- Refined dependency management by adjusting direct and indirect dependencies.

### CODE REVIEW

File: precompiles/avs/tx.go [hunk 1]
```diff
@@ -105,6 +105,14 @@ func (p Precompile) UpdateAVS(
...
 	avsParams.Action = avskeeper.UpdateAction
+	previousAVSInfo, err := p.avsKeeper.GetAVSInfo(ctx, avsParams.AvsAddress)
+	if err != nil {
+		return nil, err
+	}
+	// If avs UpdateAction check CallerAddress
+	if !slices.Contains(previousAVSInfo.Info.AvsOwnerAddress, avsParams.CallerAddress) {
+		return nil, fmt.Errorf("this caller not qualified to update %s", avsParams.CallerAddress)
+	}
 	err = p.avsKeeper.AVSInfoUpdate(ctx, avsParams)
```
Comment (bwhour) [inline precompiles/avs/tx.go]: It's better to rename this AVSInfoUpdate into UpdateAVSInfo

File: x/avs/keeper/keeper.go [hunk 2]
```diff
@@ -122,14 +122,11 @@ func (k Keeper) AVSInfoUpdate(ctx sdk.Context, params *types.AVSRegisterOrDeregi
 		if avsInfo == nil {
```
Comment (bwhour) [inline x/avs/keeper/keeper.go:123]: It's better to rename this AVSInfoUpdate into UpdateAVSInfo

File: x/dogfood/keeper/msg_server.go [hunk 3]
```diff
@@ -104,5 +108,24 @@ func (k Keeper) UpdateParams(
 		nextParams.AssetIDs = prevParams.AssetIDs
 	}
 	k.SetParams(c, nextParams)
+
+	// update the related info in the AVS module
+	isAVS, avsAddr := k.avsKeeper.IsAVSByChainID(c, avstypes.ChainIDWithoutRevision(c.ChainID()))
+	if !isAVS {
+		return nil, errors.Wrapf(types.ErrNotAVSByChainID, "chainID:%s avsAddr:%s", c.ChainID(), avsAddr)
+	}
+	err := k.avsKeeper.AVSInfoUpdate(c, &avstypes.AVSRegisterOrDeregisterParams{
```
Comment (bwhour) [inline x/dogfood/keeper/msg_server.go]: It's better to rename this AVSInfoUpdate into UpdateAVSInfo

File: x/dogfood/types/expected_keepers.go [hunk 4]
```diff
@@ -94,4 +94,5 @@ type AssetsKeeper interface {
 type AVSKeeper interface {
 	RegisterAVSWithChainID(sdk.Context, *avstypes.AVSRegisterOrDeregisterParams) (common.Address, error)
 	IsAVSByChainID(ctx sdk.Context, chainID string) (bool, common.Address)
+	AVSInfoUpdate(ctx sdk.Context, params *avstypes.AVSRegisterOrDeregisterParams) error
```
Comment (bwhour) [inline x/dogfood/types/expected_keepers.go]: It's better to rename this AVSInfoUpdate into UpdateAVSInfo

File: precompiles/avs/tx.go (hunk 1 above)
Comment (TimmyExogenous) [reply precompiles/avs/tx.go]: fixed

File: x/avs/keeper/keeper.go (hunk 2 above)
Comment (TimmyExogenous) [reply x/avs/keeper/keeper.go:123]: fixed

File: x/dogfood/keeper/msg_server.go (hunk 3 above)
Comment (TimmyExogenous) [reply x/dogfood/keeper/msg_server.go]: fixed

File: x/dogfood/types/expected_keepers.go (hunk 4 above)
Comment (TimmyExogenous) [reply x/dogfood/types/expected_keepers.go]: fixed

**Review by bwhour — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed, register a new assets in oracle and assets module, and then make a deposit and delegation to the default validator, before delegation: 
```
exocored q tendermint-validator-set
block_height: "431"
total: "1"
validators:
- address: exovalcons18z3p42xn8pjk338upvzp794h02wh7p4t7jj9jx
  proposer_priority: "0"
  pub_key:
    type: tendermint/PubKeyEd25519
    value: 8PaRnlIsW5fbLIJVv/dD+d/d162fw3ywwWcLSA0PmRQ=
  voting_power: "12765805"
```
after the delegation: 
```
exocored q assets QueOperatorAssetInfos exo18cggcpvwspnd5c6ny8wrqxpffj5zmhklprtnph
asset_infos:
  0xb82381a3fbd3fafa77b3a7be693342618240067b_0x65:
    operator_share: "0.000000000000000000"
    total_amount: "10000000000000000000"
    total_share: "10000000000000000000.000000000000000000"
    wait_unbonding_amount: "0"
  0xdac17f958d2ee523a2206206994597c13d831ec7_0x65:
    operator_share: "5000.000000000000000000"
    total_amount: "5000"
    total_share: "5000.000000000000000000"
    wait_unbonding_amount: "0"
```
sending a tx to add the new-added assets to dogfood avs support list: 
`exocored tx dogfood update-params --asset-ids "0xdac17f958d2ee523a2206206994597c13d831ec7_0x65,0xb82381a3fbd3fafa77b3a7be693342618240067b_0x65" --min-self-delegation 100 --from local_funded_account --home ~/.test-pr169 --gas-prices 7aexo`

then get operator usd value: (since the new asset price is ~$3K, delegate 10*e18 tokens, token decimal is 18), so the voting power should increase ~30K. self_usd_value is less than total usd value because the delegation is not self-delegate. 

```
exocored q operator QueryOperatorUSDValue exo18cggcpvwspnd5c6ny8wrqxpffj5zmhklprtnph 0xAa089Ba103f765fCeA44808BD3d4073523254C57
usd_values:
  active_usd_value: "12795807.515905300000000000"
  self_usd_value: "12765805.000000000000000000"
  total_usd_value: "12795807.515905300000000000"
```

then I made a self-delegation transaction, check operator usd value again:

```
exocored q operator QueryOperatorUSDValue exo18cggcpvwspnd5c6ny8wrqxpffj5zmhklprtnph 0xAa089Ba103f765fCeA44808BD3d4073523254C57
usd_values:
  active_usd_value: "12795807.515905300000000000"
  self_usd_value: "12795807.515905300000000000"
  total_usd_value: "12795807.515905300000000000"
```

**Review by cloud8little — APPROVED** [review]

File: x/dogfood/keeper/msg_server.go [hunk 5]
```diff
@@ -104,5 +108,24 @@ func (k Keeper) UpdateParams(
...
+		AssetID:           nextParams.AssetIDs,
+		UnbondingPeriod:   uint64(nextParams.EpochsUntilUnbonded),
+		MinSelfDelegation: nextParams.MinSelfDelegation.Uint64(),
+		EpochIdentifier:   nextParams.EpochIdentifier,
+		ChainID:           c.ChainID(),
+		Action:            avskeeper.UpdateAction,
+	})
+	if err != nil {
+		return nil, err
+	}
```
Comment (MaxMustermann2) [inline x/dogfood/keeper/msg_server.go:129]: Given the multiple opportunities for this to error, should we use a wrapped context? Or are the errors solely cosmetic?

**Review by MaxMustermann2 — APPROVED** [review]

File: x/dogfood/keeper/msg_server.go (hunk 5 above)
Comment (TimmyExogenous) [reply x/dogfood/keeper/msg_server.go:129]: fixed, it will return a wrapped error

**Pull request merged**